mlir/tools/mlir-tblgen/OpDocGen.cpp

```diff
@@ -381,6 +381,39 @@ static void emitAttrOrTypeDefDoc(const RecordKeeper &recordKeeper,
     emitAttrOrTypeDefDoc(AttrOrTypeDef(def), os);
 }
 
+//===----------------------------------------------------------------------===//
+// Enum Documentation
+//===----------------------------------------------------------------------===//
+
+static void emitEnumDoc(const EnumAttr &def, raw_ostream &os) {
+  os << llvm::formatv("### {0}\n", def.getEnumClassName());
+
+  // Emit the summary if present.
+  if (!def.getSummary().empty())
+    os << "\n" << def.getSummary() << "\n";
+
+  // Emit case documentation.
+  std::vector<EnumAttrCase> cases = def.getAllCases();
+  os << "\n#### Cases:\n\n";
```

Comment:
Would empty cases get flagged before here? (else we may just want to avoid emitting if there are no cases)

Reply:
I changed to not emit the cases table if there are no cases, but IMO this shouldn't be allowed by MLIR, what's the point of an enum with no cases :)

Otherwise, we could also just emit an empty table like before.

wdyt?

Reply:
Oh exactly, that was part of my question: if something would have flagged this as error already so that we don't even have to worry about the empty case. I think this is fine. doc generator can be a bit conservative :)

Reply:
As discussed offline, emitting an empty cases table also make sense since it makes it clearer that there are no cases.